packages/gatsby-source-contentful/src/download-with-retry.js

```diff
@@ -0,0 +1,34 @@
+const axios = require(`axios`)
+const rax = require(`retry-axios`)
+const { default: PQueue } = require(`p-queue`)
+
+const queue = new PQueue({
+  concurrency: 100,
+})
+
+export default async function downloadWithRetry(requestConfig, reporter) {
+  if (!requestConfig.url) {
+    throw new Error(`requestConfig.url is missing`)
+  }
+  const axiosInstance = axios.create()
+
+  axiosInstance.defaults.raxConfig = {
+    instance: axiosInstance,
+    onRetryAttempt: err => {
+      const cfg = rax.getConfig(err)
+      reporter.verbose(
+        `Retry attempt #${cfg.currentRetryAttempt} for ${requestConfig.url}`
+      )
+    },
+  }
+  rax.attach(axiosInstance)
+
+  return await queue.add(() =>
+    axiosInstance.get(requestConfig.url, {
+      ...requestConfig,
+      // This is required as we should not set `testEnvironment: "node"`
+      // in jest.config.js just to test this properly
+      adapter: require(`axios/lib/adapters/http`),
```

Comment:
If you have an idea to set a jest configuration just for this plugin, lets do that. But as we run jest on the projects root directory, I found to way to set this just for the plugin.

See: https://github.com/axios/axios/issues/305#issuecomment-602032195

Reply:
You can use 

```
/**
 * @jest-environment node
 */